drivers/ieee802154/ieee802154_cc2520.c

```diff
@@ -1283,7 +1283,7 @@ static int _cc2520_crypto_ccm(struct cipher_ctx *ctx,
 		return -EINVAL;
 	}
 
-	if (apkt->tag) {
+	if (!apkt->tag) {
```

Comment:
Just remove that if() altogether.

Reply:
@tbursztyka  Does I miss something for CC2520, could you please help explain more so that I can change the code better? Thanks!

Reply:
Just remove that if () and its code block.

On ciphering phase: cc2520 is going to generate the tag, but is not going to give it back to the caller (the caller sets tag to NULL anyway).
On deciphering phase: cc2520 will need a tag depending on security level configuration

conclusion: there is no need to check tag pointer.

Reply:
@tbursztyka  Thanks!
And do we also need remove below memcpy? If remove, it doesn't behave as our doc described.

Reply:
let me answer at the right line

Reply:
@tbursztyka  Thanks.